### PR TITLE
for issue #12 (# 51) (rebased on current master)

### PR DESCRIPTION
**Pull request by jwaldmann**
working mailman parser

attach (default) date information so that feed can be parsed by importer

prepare the date hack

annotate items with dates

### CODE REVIEW

**Comment by lwm** [comment]
@chrisdone, can you point the Travis build to the organization repo?

**Comment by lwm** [comment]
Thanks @jwaldmann! I'll give it a review and hopefully we get the Travis build back pretty soon so we can give it a run :)

File: src/HN/Model/Mailman.hs [hunk 1]
```diff
@@ -0,0 +1,137 @@
+-- | construct a feed from a mailman (pipermail) archive,
+-- e.g., https://mail.haskell.org/pipermail/libraries/
```
Comment (lwm) [inline src/HN/Model/Mailman.hs]: Probably not necessary, we have a search function :+1:

File: src/HN/Model/Mailman.hs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+-- see https://github.com/chrisdone/haskellnews/issues/12
+-- and https://github.com/haskell-infra/hl/issues/63#issuecomment-74420850
+
+{-# language OverloadedStrings #-}
+{-# language BangPatterns #-}
+
+module HN.Model.Mailman where
+
+import Network.HTTP.Conduit
+-- I tried with
```
Comment (lwm) [inline src/HN/Model/Mailman.hs]: Could you open an issue with the traces / or further information on why it wouldn't work? 

We could maybe see about fixing it.

Then you can remove this comment as well.

File: src/HN/Model/Mailman.hs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import Data.Char (isDigit)
+
+import System.IO
+
+test1 = do
+  downloadFeed 10 "https://mail.haskell.org/pipermail/haskell-cafe/"
+
+
+-- | look at archive and produce a feed.
+-- 1st argument is minimal number of items to produce.
```
Comment (lwm) [inline src/HN/Model/Mailman.hs]: Could use type synonyms here to encode your description?

File: src/HN/Model/Mailman.hs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      -- mapM_ print items
+      let feed0 = newFeed (RSSKind Nothing)
+          feed = id
+               $ withFeedPubDate pubdate
+               $ foldr addItem feed0 items
+      -- print feed    
+      return $ Right feed
+
+getDoc uri = do
+  -- hPutStrLn stderr $ unwords ["getDoc", uri]
```
Comment (lwm) [inline src/HN/Model/Mailman.hs]: Debug statements should go?

File: src/HN/Model/Mailman.hs [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                  >>= mkItem uri (dropSuffix "thread.html" d ) 
+      later <- getItems (its - length items) uri ates
+      return $ items ++ later
+
+dropSuffix suf s =
+  if T.isSuffixOf suf s
+  then T.take (T.length s - T.length suf) s
+  else s
+
+fetch_date cursor desc = do
```
Comment (lwm) [inline src/HN/Model/Mailman.hs]: fetchDate?

File: src/HN/Model/Mailman.hs (hunk 1 above)
Comment (jwaldmann) [inline src/HN/Model/Mailman.hs]: (But I like motativational text at the beginning of code.)

File: src/HN/Model/Mailman.hs (hunk 2 above)
Comment (jwaldmann) [inline src/HN/Model/Mailman.hs]: Later.

File: src/HN/Model/Mailman.hs (hunk 4 above)
Comment (jwaldmann) [inline src/HN/Model/Mailman.hs]: OK.

File: src/HN/Model/Mailman.hs (hunk 5 above)
Comment (jwaldmann) [inline src/HN/Model/Mailman.hs]: OK.

File: src/HN/Model/Mailman.hs (hunk 3 above)
Comment (jwaldmann) [inline src/HN/Model/Mailman.hs]: Type synonyms don't help much - newtypes are better, but I don't care enough (in this case). HN.Model.Feeds.importGeneric uses String for URI as well.

**Comment by jwaldmann** [comment]
(bump) What is the status here? I don't quite see which of your requested changes do you think are blocking the merge. (I think: none.)

**Comment by lwm** [comment]
Thanks for taking the time on this @jwaldmann!

**Pull request merged**